src/contracts/proxies/ODSafeManager.sol

```diff
@@ -66,6 +66,7 @@ contract ODSafeManager is IODSafeManager {
   constructor(address _safeEngine, address _vault721) {
     safeEngine = _safeEngine.assertNonNull();
     vault721 = IVault721(_vault721);
+    _safeId = vault721.totalSafes();
```

Comment:
This does not address the underlying problem due to the way that governance works. There will be a delay of multiple days between when `ODSafeManager` is deployed, and when the governance process is completed to complete the upgrade to the new `ODSafeManager`

Reply:
@pi0neerpat sorry about that - still getting acclimated to your process I suppose.  I understand completely now.

# Single Truthful Counter

`_safeId` is only incremented during calls to `openSAFE` on the ODSafeManager.  `openSAFE` then calls into the Vault721 `mint` function; which is access control - only callable by the ODSafeManager.  Since the ODSafeManager is designed to be effectively upgradeable by governance - it doesn't make sense for it to maintain it's own independent counter.  If we have two independent counters that must be sync'd at all times ... in my mind that's a recipe for unintended behavior.

Since the Vault721 is designed to be an immutable hyper structure - it makes sense to move the `safeId` counter into the Vault721 contract; and access it directly from the `ODSafeManager` during calls to `openSafe`.

## Solution

Some of the requirements have already been implemented above; such as a simple interface for accessing the `totalSafes` state in the Vault721 contract and it's corresponding view function and counter.

Remove `safeId` from the ODSafeManager; and instead access the `totalSafes` directly from Vault721 during calls to `openSAFE`:

```
/// @inheritdoc IODSafeManager
  function openSAFE(bytes32 _cType, address _usr) external returns (uint256 _id) {
    if (_usr == address(0)) revert ZeroAddress();

    _safeId = vault721.totalSafes();
    ++_safeId;
    address _safeHandler = address(new SAFEHandler(safeEngine));

    _safeData[_safeId] = SAFEData({owner: _usr, safeHandler: _safeHandler, collateralType: _cType});

    // Save the address of the safeHandler
    handlerExists[_safeHandler] = true;

    _usrSafes[_usr].add(_safeId);
    _usrSafesPerCollat[_usr][_cType].add(_safeId);

    vault721.mint(_usr, _safeId);

    emit OpenSAFE(msg.sender, _usr, _safeId);
    return _safeId;
  }
  ```
  
  This of course also requires removing the incrementation from `mint`; and it turns out you were 100% right we don't need to enforce the _safeId matches the counter because the function is access control:
  
  ```
    /**
   * @dev mint can only be called by the SafeManager
   * enforces that only ODProxies call `openSafe` function by checking _proxyRegistry
   */
  function mint(address _proxy, uint256 _safeId) external {
    require(msg.sender == address(safeManager), 'V721: only safeManager');
    require(_proxyRegistry[_proxy] != address(0), 'V721: non-native proxy');
    totalSafes == _safeId;
    address _user = _proxyRegistry[_proxy];
    _safeMint(_user, _safeId);
  }
  ```

Reply:
Discussion in Discord

Reply:
New migration strategy discussion can be found here: https://github.com/open-dollar/od-contracts/issues/323#issuecomment-1880079499